### PR TITLE
fix(server.set): it should not be possible to set a connected server

### PR DESCRIPTION
**Pull request by badrAZ**

### CODE REVIEW

File: src/api/server.js [hunk 1]
```diff
@@ -69,6 +70,13 @@ getAll.permission = 'admin'
 // -------------------------------------------------------------------
 
 export async function set ({id, ...props}) {
+  if (find(
+    await this.getAllXenServers(),
+    { id }
+  ).status === 'connected') {
+    throw new Error('you can only set the server when it is disconnected.')
+  }
```
Comment (julien-f) [inline src/api/server.js]: Update `updateXenServer()` instead.

Comment (badrAZ) [inline src/api/server.js]: `updateXenServer()` is used  also in `connectXenServer()` to update the error property.

Comment (julien-f) [inline src/api/server.js]: If you just do the check when updating certain entries it should be fine.

File: src/xo-mixins/xen-servers.js [hunk 2]
```diff
@@ -98,6 +99,18 @@ export default class {
...
+      host !== undefined ||
+      password !== undefined ||
+      readOnly !== undefined ||
+      username !== undefined
+
+    if (
+      needDisconnectTheServer &&
+      find(await this.getAllXenServers(), { id }).status === 'connected'
+    ) {
+      throw new Error('this entry require disconnecting the server to update it.')
```
Comment (julien-f) [inline src/xo-mixins/xen-servers.js]: No leading uppercase and no trailing period in error message.

File: src/xo-mixins/xen-servers.js [hunk 3]
```diff
@@ -98,6 +99,18 @@ export default class {
     username
   }) {
     const server = await this._getXenServer(id)
+    const needDisconnectTheServer = allowUnauthorized !== undefined ||
```
Comment (julien-f) [inline src/xo-mixins/xen-servers.js]: `needDisconnectTheServer` → `requireDisconnected`

Comment (julien-f) [inline src/xo-mixins/xen-servers.js]: Move `allowUnauthorized !== undefined` on the next line so it's at the same level of other entries.

File: src/xo-mixins/xen-servers.js [hunk 4]
```diff
@@ -98,6 +99,18 @@ export default class {
...
+      host !== undefined ||
+      password !== undefined ||
+      readOnly !== undefined ||
+      username !== undefined
+
+    if (
+      needDisconnectTheServer &&
+      find(await this.getAllXenServers(), { id }).status === 'connected'
```
Comment (julien-f) [inline src/xo-mixins/xen-servers.js]: No need to use either `find` or `getAllXenServers`.

File: src/xo-mixins/xen-servers.js [hunk 5]
```diff
@@ -98,6 +98,21 @@ export default class {
     username
   }) {
     const server = await this._getXenServer(id)
+    const xapi = this._xapis[id]
+    const requireDisconnected =
+      allowUnauthorized !== undefined ||
+      host !== undefined ||
+      password !== undefined ||
+      readOnly !== undefined ||
```
Comment (julien-f) [inline src/xo-mixins/xen-servers.js]: Can be changed while connected.

**Pull request merged**